### PR TITLE
[EP-864] Added Mokingjay and Cocoapods

### PR DESCRIPTION
**Pull request by AndreiGalkin**

### CODE REVIEW

File: Source/Services/CardValidation/JPCardValidationService.m [hunk 1]
```diff
@@ -407,23 +407,23 @@ - (JPValidationResult *)validateUSAPostalCodeInput:(NSString *)input {
 
 - (JPValidationResult *)validateCanadaPostalCodeInput:(NSString *)input {
 
-    if (input.length > 6) {
+    if (input.length > 8) {
```
Comment (mpetrenco) [inline Source/Services/CardValidation/JPCardValidationService.m:410]: Why 8? 
Canada went from `A1A1A1` to `A1A 1A1`. That's one extra character.

File: Tests/APITests/ApplePayTests.swift [hunk 2]
```diff
@@ -43,7 +43,7 @@ class ApplePayTests: JudoTestCase {
                                               merchantId: "",
                                               currency: "GBP",
                                               countryCode: "GB",
-                                              paymentSummaryItems: items)
+                                              JPPaymentSummaryItems: items)
```
Comment (mpetrenco) [inline Tests/APITests/ApplePayTests.swift:46]: Please revert this

File: Tests/APITests/ApplePayTests.swift [hunk 3]
```diff
@@ -104,7 +104,7 @@ class ApplePayTests: JudoTestCase {
      *         its "pkPaymentAuthorizationViewController" instance method
      */
     func test_OnInvalidConfiguration_ManagerReturnsNilPKAuthorizationViewController() {
-        configuration.paymentSummaryItems = [];
+        configuration.JPPaymentSummaryItems = [];
```
Comment (mpetrenco) [inline Tests/APITests/ApplePayTests.swift:107]: Please revert this

File: Tests/APITests/IDEALManagerTests.swift [hunk 4]
```diff
@@ -26,11 +26,11 @@ import XCTest
 
 class IDEALManagerTests: JudoTestCase {
     
-    private var idealManager: JPIDEALService!
+    private var idealManager: JPIDealService!
```
Comment (mpetrenco) [inline Tests/APITests/IDEALManagerTests.swift:29]: Please revert to `JPIDEALServicee` to keep consistency

File: Source/Services/CardValidation/JPCardValidationService.m [hunk 5]
```diff
@@ -362,7 +362,7 @@ - (JPValidationResult *)validateUKPostalCodeInput:(NSString *)input {
         return self.lastPostalCodeValidationResult;
     }
 
-    NSString *UKRegexString = @"(GIR 0AA)|((([A-Z-[QVX]][0-9][0-9]?)|(([A-Z-[QVX]][A-Z-[IJZ]][0-9][0-9]?)|(([A-Z-[QVX‌​]][0-9][A-HJKSTUW])|([A-Z-[QVX]][A-Z-[IJZ]][0-9][ABEHMNPRVWXY]))))\\s?[0-9][A-Z-[C‌​IKMOV]]{2})";
+    NSString *const UKRegexString = @"(GIR 0AA)|((([A-Z-[QVX]][0-9][0-9]?)|(([A-Z-[QVX]][A-Z-[IJZ]][0-9][0-9]?)|(([A-Z-[QVX‌​]][0-9][A-HJKSTUW])|([A-Z-[QVX]][A-Z-[IJZ]][0-9][ABEHMNPRVWXY]))))\\s?[0-9][A-Z-[C‌​IKMOV]]{2})";
```
Comment (mpetrenco) [inline Source/Services/CardValidation/JPCardValidationService.m:365]: Small detail but constants should have a prefix like so: `kUKRegexString` to keep consistency

File: Tests/Modules/Sevices/JPTransactionServiceTest.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    func testTransactionWithConfigurations() {
+        let transaction = transactionService.transaction(with: self.transactionConfigurations)
+        XCTAssertEqual(transaction.amount!.amount, self.transactionAmmount.amount)
+        XCTAssertEqual(transaction.amount!.currency, self.transactionAmmount.currency)
+        XCTAssertEqual(transaction.reference!.consumerReference, self.transactionReference.consumerReference)
+        XCTAssertEqual(transaction.reference!.paymentReference, self.transactionReference.paymentReference)
+    }
+
+    func testListTransactionsOfType() {
+        self.transactionService.listTransactions(of: .checkCard,
```
Comment (mpetrenco) [inline Tests/Modules/Sevices/JPTransactionServiceTest.swift:46]: Might consider replacing this with a Mockingjay alternative, as to not make real network calls.

File: Tests/Modules/Sevices/JPTransactionServiceTest.swift [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                                                  paginated: JPPagination(offset: 1,
+                                                                          pageSize: 2,
+                                                                          sort: ""),
+                                                  completion: { (response, error) in
+                                                    XCTAssertNotNil(response)
+        })
+    }
+
+    func testreceiptForReceiptId(){
```
Comment (mpetrenco) [inline Tests/Modules/Sevices/JPTransactionServiceTest.swift:55]: `testreceiptForReceiptId` -> `testReceiptForReceiptId `

File: Tests/Modules/Sevices/JPTransactionServiceTest.swift (hunk 6 above)
Comment (AndreiGalkin) [reply Tests/Modules/Sevices/JPTransactionServiceTest.swift:46]: I'll replace all network calls with mocks in next PR.

**Pull request merged**